test/replication-luatest/gh_8746_replica_transaction_boundary_test.lua

```diff
@@ -0,0 +1,124 @@
+local t = require('luatest')
+local server = require('luatest.server')
+local replica_set = require('luatest.replica_set')
+
+local g = t.group('gh-8746-transaction-boundaries')
+
+g.before_each(function(cg)
+    cg.replica_set = replica_set:new{}
+end)
+
+g.after_each(function(cg)
+    cg.replica_set:drop()
+end)
+
+local function prepare(cg)
+    cg.replica_set:start()
+    cg.master:exec(function()
```

Comment:
The test is flaky. Orphan state:
```
[001] replication-luatest/gh_8746_replica_transactio>                 [ fail ]
[001] Test failed! Output from reject file /tmp/t/rejects/replication-luatest/gh_8746_replica_transaction_boundary.reject:
[001] Tarantool version is 3.0.0-alpha2-38-g5c6d83159
[001] TAP version 13
[001] 1..3
[001] # Started on Thu Aug 31 15:49:21 2023
[001] # Starting group: gh-8746-transaction-boundaries
[001] ok     1	gh-8746-transaction-boundaries.test_replica_recovery
[001] not ok 2	gh-8746-transaction-boundaries.test_replication_to_master
[001] #   Can't modify data on a read-only instance - it is an orphan
[001] #   stack traceback:
[001] #   	...on-luatest/gh_8746_replica_transaction_boundary_test.lua:17: in function 'prepare'
[001] #   	...on-luatest/gh_8746_replica_transaction_boundary_test.lua:77: in function 'gh-8746-transaction-boundaries.test_replication_to_master'
[001] #   	...
[001] #   	[C]: in function 'xpcall'
[001] #   artifacts:
[001] #   master -> /tmp/t/001_replication-luatest/artifacts/rs-7LOVjZBt2xAa/master-szUBIBOHrp_4
[001] #   replica -> /tmp/t/001_replication-luatest/artifacts/rs-7LOVjZBt2xAa/replica-MKQXDNdzjZy5
[001] ok     3	gh-8746-transaction-boundaries.test_replica_force_recovery
[001] # Ran 3 tests in 3.281 seconds, 2 succeeded, 1 errored
```

Reply:
Thanks for noticing!

Doesn't flake on my machine, so I can't check, but hope this helps:
```diff
diff --git a/test/replication-luatest/gh_8746_replica_transaction_boundary_test.lua b/test/replication-luatest/gh_8746_replica_transaction_boundary_test.lua
index d0a2f73dfb..4fac79bb51 100644
--- a/test/replication-luatest/gh_8746_replica_transaction_boundary_test.lua
+++ b/test/replication-luatest/gh_8746_replica_transaction_boundary_test.lua
@@ -15,6 +15,7 @@ end)
 local function prepare(cg)
     cg.replica_set:start()
     cg.master:exec(function()
+        box.ctl.wait_rw()
         box.schema.space.create('test')
         box.space.test:create_index('pk')
         box.schema.space.create('loc', {is_local = true})
```